prog/rand.go

```diff
@@ -805,6 +821,15 @@ func (a *UnionType) generate(r *randGen, s *state, dir Dir) (arg Arg, calls []*C
 }
 
 func (a *PtrType) generate(r *randGen, s *state, dir Dir) (arg Arg, calls []*Call) {
+	switch t := a.Elem.(type) {
+	case *BufferType:
```

Comment:
This is misplaced. Generation of pointer should not generate all possible pointee types. Having just this one here as a special case is wrong. It should be generated by BufferType generator.

Reply:
with this code block, I'm getting this in corpus:

```
r0 = openat$sysfs(0xffffffffffffff9c, &(0x7f0000000bc0)**=''**, 0x103241, 0x0)
```

Reply:
Without this code block, I'm getting this in corpus:
```
r0 = openat$sysfs(0xffffffffffffff9c, &(0x7f0000000bc0)='', 0x103241, 0x0)
```
while I'm expecting it to be:
```
r0 = openat$sysfs(0xffffffffffffff9c, &(0x7f0000000bc0)='/sys/xxx', 0x103241, 0x0)
```

Reply:
This and some other things are due to bug in isDefaultArg implementation.